src/NRules/NRules.Fluent/Expressions/QueryExpression.cs

```diff
@@ -32,6 +32,17 @@ public void FactQuery<TSource>(Expression<Func<TSource, bool>>[] conditions)
             };
         }
 
+        public void From<TSource>(Expression<Func<TSource>> source)
+        {
+            _buildAction = (b, n) =>
+            {
+                var patternBuilder = b.Pattern(typeof(TSource), n);
+                var bindingBuilder = patternBuilder.Binding();
+                bindingBuilder.DslBindingExpression(_groupBuilder.Declarations, source);
+                return patternBuilder;
+            };
+        }
```

Comment:
I'm wondering if adding a Binding here is really needed. Your intent here is not to create a new binding, but to create a new query off of an existing one. Will this work if you just find the existing pattern that `source` refers to, and return that?

Reply:
Hmm what if the user wants to do something like

```csharp
When()
    .Query(() => firstQuery, q => q ...)
    .Let(() => someVariable, () => ... )
    .Query(() => secondQuery, q => q
        .From(() => SomeFunctionOn(firstQuery, someVariable))
        .GroupBy(() => ...))
```

i.e. it may actually require a full binding expression.

I agree if we only expect users to put existing queries in `From`, then we can just find the existing pattern. (coincidentally, what's the right way to do that?)

Reply:
Like in 82d48a7

Reply:
Interesting.. I was thinking of this as just ability to build a query on top of an existing pattern. But what you are doing here could be more powerful. On the other hand, this could be implemented by taking an existing pattern, and then following it by a `Select` that could call that function. I'm going to accept this PR, and we can loop on this.